assets/js/components/ProjectEditorTextField.js

```diff
@@ -0,0 +1,175 @@
+  this.pathEditDescription = projectDescriptionCredits.data('path-edit-program-description')
+  this.pathEditCredits = projectDescriptionCredits.data('path-edit-program-credits')
+
+  this.setText = () => {
+    this.editText.val(this.initialText)
+    this.lastSavedText = this.initialText
+  }
+
+  this.areChangesSaved = () => {
+    return this.editText.val().trim() === this.lastSavedText.trim()
+  }
+
+  this.save = (languageSelected) => {
+    hideError()
+    const newText = this.editText.val().trim()
+    if (this.areChangesSaved() || this.shouldDisable(languageSelected)) {
+      // Return without saving
+    } else if (languageSelected === '' || languageSelected === 'default') {
+      let url
+      if (this.programSection === 'name') {
+        url = this.pathEditName
+      } else if (this.programSection === 'description') {
+        url = this.pathEditDescription
+      } else if (this.programSection === 'credits') {
+        url = this.pathEditCredits
+      }
+
+      return new Promise(function (resolve, reject) {
+        $.ajax({
+          url: url,
+          type: 'put',
+          data: { value: newText },
+          success: function (data) {
+            const statusCode = parseInt(data.statusCode)
+            if (statusCode === 527 || statusCode === 707) {
+              showError(data)
+              reject(statusCode)
+            } else {
+              resolve(statusCode)
+            }
+          },
+          error: function (error) {
+            if (error.status === 422) {
+              error.message = error.responseText
+              showError(error)
+              reject(error.status)
+            } else if (error.status === 401) {
+              reject(error.status)
+            }
+          }
+        })
+      })
+    } else if (newText === '') {
+      return this.customTranslationApi.deleteCustomTranslation(
+        this.programId,
+        languageSelected,
+        hideError,
+        showError
+      )
+    } else {
+      return this.customTranslationApi.saveCustomTranslation(
+        this.programId,
+        newText,
+        languageSelected,
+        hideError,
+        showError
+      )
+    }
+  }
+
+  this.getNewText = (language) => {
+    hideError()
+    if (language === 'default') {
+      this.editText.removeAttr('disabled')
+      this.editText.val(this.initialText)
+      this.lastSavedText = this.initialText
+      return
+    } else if (this.shouldDisable(language)) {
+      this.lastSavedText = this.initialText
+      this.editText.val(this.initialText)
+      this.editText.attr('disabled', '')
+      return
+    }
+
+    this.editText.attr('disabled', '')
+    this.editText.val('')
+    this.textLoadingSpinner.show()
+
+    this.customTranslationApi.getCustomTranslation(
+      this.programId,
+      language,
+      getCustomTranslationSuccess,
+      getCustomTranslationError
+    )
+  }
+
+  this.keepChanges = (language) => {
```

Comment:
this name is very confusing. the function is very similar to getNewText, except it doesn't touch the entered text in the editor. so maybe a name like getNewTextKeepChanges?

Reply:
Renamed